packages/grafana-ui/src/utils/measureText.ts

```diff
@@ -43,3 +51,19 @@ export function calculateFontSize(text: string, width: number, height: number, l
   const optimalSize = Math.min(fontSizeBasedOnHeight, fontSizeBasedOnWidth);
   return Math.min(optimalSize, maxSize ?? optimalSize);
 }
+
+/**
+ * @beta
+ */
+export function getCanvasGradient(color: string, amount: number, plotHeight: number): CanvasGradient {
+  const ctx = getCanvasContext();
+  var gradient = ctx.createLinearGradient(0, 0, 0, plotHeight);
```

Comment:
hmm, what happens to this when `plotHeight` changes due to resize? i guess just live with it or kill perf and have some form of throttle/debounce that updates config and rehinits the plot?

technically, you can just mutate `u.series[i].fill` and invoke a cheap cached `u.redraw(false)`.

tricky.

Reply:
think we need to pass in the current graph height here, so graph height change will need to invalidate/update the config.

Reply:
right, but chart resize happens every dragged pixel? and changes in config cause expensive plot recreation. so this might need to be throttled till the resize ends?